Added explicit default for ASDF_DATA_DIR environment variable

log2 opened the pull request:
Fix for #6

Commit:
Added explicit default for ASDF_DATA_DIR environment variable, aligned with defaults defined in asdf

andrewthauer reviewed (approved):
Makes sense, thanks!

It's been a while since I've looked at the core `asdf` code base, but seems it seems unnecessary that plugins should need to do this in the first place 🤔. Another approach could be to have core set the default before invoking plugin commands. But again, I couldn't say this makes sense for sure without digging into core more.

log2 commented:
@andrewthauer , you're right, absolutely! Plugin asdf alias could use library function `asdf_data_dir` (see https://github.com/asdf-vm/asdf/blob/master/lib/utils.bash#L39. I'll try to use it soon, then resubmit another updated PR.

Commit:
Updated in order to use asdf_data_dir library function

Note that, due to the way asdf works with plugins, bash script file has to be non-executable in order for it to be sourced together with $ASDF_DIR/lib/utils.bash, thus having all asdf library functions available.

This commit handles comment https://github.com/andrewthauer/asdf-alias/pull/7#pullrequestreview-885590821

log2 commented:
@andrewthauer , please, have a look at this revised PR: I have switched from a direct reference to environment variable to a call to the asdf's library function `asdf_data_dir`, as anticipated. Also, in order for all library functions from `$ASDF_DIR/lib/utils.bash` to be available while sourcing `command.bash`, I have disabled execute bit on the file. It would be also feasible to have `$ASDF_DIR/lib/utils.bash` sourced explicitly at the very start of the file. Would you prefer the latter?

andrewthauer commented on lib/commands/command.bash:
```diff
@@ -135,7 +135,7 @@ main() {
   # alias="$2" -- not used directly
   version="$3"
-  install_dir="$ASDF_DATA_DIR/installs/$plugin"
+  install_dir="${asdf_data_dir)/installs/$plugin"
```
```suggestion
  install_dir="$(asdf_data_dir)/installs/$plugin"
```

log2 replied:
Thank you @andrewthauer for catching that nasty typo, I have duly noted and fixed as per your commit suggestion.

Speaking about executable state of script, it seems necessary to remove executable bit, since you get two different behaviours depending on it (one of these results in asdf alias not working properly). Let me show this.

- *Case 1: command.bash is executable*

In this case, command.bash is on its own for sourcing any dependent library, such as the one providing function `asdf_data_dir`; since no loading of such library happens, when you try calling asdf alias you get:

```
❯ chmod +x <asdf-dir>/plugins/alias/lib/commands/command.bash
❯ asdf alias
<asdf-dir>/plugins/alias/lib/commands/command.bash: line 138: asdf_data_dir: command not found
```

-  *Case 2: command.bash is not executable (but it still being valid sourceable bash code)*

In this case, command.bash is sourced directly while `asdf` command is executed; all asdf libraries are available, including the one providing `asdf_data_dir` function (in fact, `<asdf-dir>/lib/utils.bash` is source here https://github.com/asdf-vm/asdf/blob/master/bin/asdf#L4 ).

So you get:

```
❯ chmod -x <asdf-dir>/plugins/alias/lib/commands/command.bash
❯ asdf alias
usage
  asdf alias <plugin> <name> [<version> | --auto | --remove]
  asdf alias <plugin> --auto
  asdf alias <plugin> [--list] 
```

I have tested this locally, both in bash and in zsh, by directly modifying my local copy of asdf-alias code.

Would you like to test it, too, in order to check whether you get the same results? Thank you

andrewthauer reviewed (commented):
@log2 - Thanks for checking. You suggested change seems good aside from a small typo in executing the command.

Based on the [plugin docs for custom commands](https://asdf-vm.com/plugins/create.html#extension-commands-for-asdf-cli) it sounds like both scripts and executables are acceptable so makes sense to make this a script vs executable. With this it seems unnecessary to source the utils file directly (based on my testing).

log2 committed:
Fixed typo in modified lib/commands/command.bash

Co-authored-by: Andrew Thauer <athauer@hotmail.com>

andrewthauer commented:
@log2 - I confirmed the other day this execute bit flip would be required and also works on my end. Thanks for the change!
